cf-check/diagnose.c

```diff
@@ -155,19 +172,33 @@ int diagnose_main(int argc, const char *const *const argv)
 {
     size_t offset = 1;
     bool foreground = false;
-    if (StringSafeEqual(argv[1], "--no-fork")
-        || StringSafeEqual(argv[1], "-F"))
+    bool validate = false;
+    for (int i = offset; i < argc && argv[i][0] == '-'; ++i)
     {
-        foreground = true;
-        offset += 1;
+        if (StringMatchesOption(argv[i], "--no-fork", "-F"))
+        {
+            foreground = true;
+            offset += 1;
```

Comment:
`offset += 1` almost looks suspicious to me. As if there was supposed to be something else than `1` because otherwise `++` would be used, right?

Reply:
No, I don't agree, I just prefer `+= 1`. More readable, less cryptic, IMO.